### PR TITLE
Remove deprecated features for 0.14.

### PR DESCRIPTION
**Pull request by bdice**
## Description
- Remove deprecated `flow.run`.
- Remove `walltime` argument and CLI option; update template tests.
- Remove deprecated `FlowProject.export_job_statuses`.
- Remove deprecated `script` feature (use `submit --pretend` instead).
- Remove deprecated `CPUEnvironment`, `GPUEnvironment`.

## Types of Changes

- [x] Documentation update
- [ ] Bug fix
- [x] New feature
- [x] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/environment.py [hunk 1]
```diff
@@ -225,12 +223,7 @@ def add_args(cls, parser):
             The argument parser where arguments will be added.
 
         """
-        parser.add_argument(
-            "-w",
-            "--walltime",
-            type=float,
-            help="The wallclock time in hours. Deprecated in 0.13 and will be removed in 0.14.",
-        )
+        pass
```
Comment (mikemhenry) [inline flow/environment.py:226]: Why are we keeping this method?

**Review by mikemhenry — APPROVED** [review]
LGTM! Just had one question

File: flow/environment.py (hunk 1 above)
Comment (bdice) [reply flow/environment.py:226]: This method is called by all the children of ComputeEnvironment. This has been an empty method for a long time, I just changed it in #476 as a temporary workaround until it was removed in this PR.

Comment (mikemhenry) [reply flow/environment.py:226]: Got it!

**Pull request merged**